RUFAS/data_structures/animal_to_manure_connection.py

```diff
@@ -0,0 +1,217 @@
+    PARLOR: int
+        Represents manure from the parlor.
+    GENERAL: int
+        Represents all manure other than what is deposited in or traveling to/from the milking parlor.
+    """
+
+    PARLOR = 0
+    GENERAL = 1
+
+
+@dataclass
+class PenManureData:
+    """
+    Information about the pen or pens in which the manure was produced.
+
+    Attributes
+    ----------
+    num_animals : int
+        The number of animals in this pen that created the manure.
+    manure_deposition_surface_area : float
+        The surface area of the manure deposition area in the pen (m^2).
+    animal_combination : AnimalCombination
+        The combination of animals in the pen.
+    pen_type : str | None
+        The type of pen.
+    manure_urine_mass : float
+        The overall mass of urine in the manure stream (kg).
+    manure_urine_nitrogen : float
+        The mass of nitrogen in the urine in the manure stream (kg).
+    stream_type : ManureStreamType
+        The type of manure stream in the pen.
+    """
+
+    num_animals: int
+    manure_deposition_surface_area: float
+    animal_combination: AnimalCombination
+    pen_type: str | None
+    manure_urine_mass: float
+    manure_urine_nitrogen: float
+    stream_type: ManureStreamType
+
+    def __post_init__(self) -> None:
+        if self.stream_type == ManureStreamType.PARLOR and self.animal_combination != AnimalCombination.LAC_COW:
+            raise ValueError("Parlor manure stream must be from a lactating cow pen.")
+
+    def __add__(self, other: "PenManureData") -> "PenManureData":
+        """
+        Combines two PenManureData instances.
+
+        Parameters
+        ----------
+        other : PenManureData
+            The other PenManureData instance to combine with this one.
+
+        Returns
+        -------
+        PenManureData
+            The combined PenManureData instance.
+
+        Raises
+        ------
+        ValueError
+            If the stream type is ManureStreamType.GENERAL or if the animal combinations do not match.
+
+        """
+        if self.stream_type == ManureStreamType.GENERAL or other.stream_type == ManureStreamType.GENERAL:
+            raise ValueError("Cannot combine PenManureData instances with a general manure stream type.")
+        if self.animal_combination != other.animal_combination:
+            raise ValueError("Cannot combine PenManureData instances with different animal combinations.")
+
+        return PenManureData(
+            num_animals=self.num_animals + other.num_animals,
+            manure_deposition_surface_area=self.manure_deposition_surface_area + other.manure_deposition_surface_area,
+            animal_combination=self.animal_combination,
+            pen_type=None,
+            manure_urine_mass=self.manure_urine_mass + other.manure_urine_mass,
+            manure_urine_nitrogen=self.manure_urine_nitrogen + other.manure_urine_nitrogen,
+            stream_type=self.stream_type,
+        )
+
+
+class ManureStream:
```

Comment:
```suggestion

@dataclass
class ManureStream:
```
Shoot I missed this when I reviewed ManureStream, that is my bad. Would you please convert this to a dataclass?